internal/observability/events/eventer_retry.go

```diff
@@ -0,0 +1,79 @@
+package event
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/eventlogger"
+	"github.com/hashicorp/go-multierror"
+)
+
+const (
+	// stdRetryCount is the standard number of times for retry when sending events
+	stdRetryCount = 3
+)
+
+type backoff interface {
+	duration(attemptNumber uint) time.Duration
+}
+
+type expBackoff struct{}
+
+func (b expBackoff) duration(attempt uint) time.Duration {
+	r := rand.Float64()
+	return time.Millisecond * time.Duration(math.Exp2(float64(attempt))*5*(r+0.5))
+}
+
+type retryInfo struct {
+	retries int
+	backoff time.Duration
+}
+
+type sendHandler func() (eventlogger.Status, error)
+
+func (e *Eventer) retrySend(ctx context.Context, retries uint, backOff backoff, handler sendHandler) error {
+	const op = "event.(Eventer).retrySend"
+	if backOff == nil {
+		return errors.New(errors.InvalidParameter, op, "missing backoff")
+	}
+	if handler == nil {
+		return errors.New(errors.InvalidParameter, op, "missing handler")
+	}
+	success := false
+	var retryErrors error
+	var attemptStatus eventlogger.Status
+	info := retryInfo{}
+	for attempts := uint(1); ; attempts++ {
+		if attempts > retries+1 {
+			retryErrors = multierror.Append(retryErrors, errors.New(errors.MaxRetries, op, fmt.Sprintf("Too many retries: %d of %d", attempts-1, retries+1)))
```

Comment:
This is probably fine and me being super nitty, but it seems strange to call `retrySend` with the retries parameter as `3` and then get an error message: `Too many retries: 4 of 4`

Reply:
if retries == 3 is it failed after 3 attempts or failed after 4 attempts? does the UX imply that if 0 is passed for retries that no attempt will be made or just 1? This is what I was trying to disambiguate unsuccessfully I guess. At this point, I'm punting on even trying to conveue the actual number of attempts and just saying being as vauge as possible and returning the reached max of %d where %d is the number of retries.